src/monattack.cpp

```diff
@@ -4649,17 +4649,19 @@ bool mattack::dodge_check( monster *z, Creature *target ) {
     return false;
 }
 
-void mattack::lay_egg( monster *z )
+bool mattack::lay_egg( monster *z )
 {
     monster &critter = *z;
     const furn_t &furn = g->m.furn( critter.pos() ).obj();
 
-    if( !critter.is_dead() && critter.has_flag( MF_EGGLAYING ) && furn.has_flag( TFLAG_NEST ) &&
-        !critter.has_effect( effect_laid_egg ) ) {
+    if( !critter.is_dead() && furn.has_flag( TFLAG_NEST ) ) {
```

Comment:
You can skip the dead check - it is already checked before executing the attack function.
But you do need to check the laid egg effect - this one won't be checked automatically.